src/main/java/baritone/process/MineProcess.java

```diff
@@ -319,6 +319,21 @@ public double heuristic(int x, int y, int z) {
             int zDiff = z - this.z;
             return GoalBlock.calculate(xDiff, yDiff < -1 ? yDiff + 2 : yDiff == -1 ? 0 : yDiff, zDiff);
         }
+
+        @Override
+        public boolean equals(Object o) {
+            return super.equals(o);
+        }
```

Comment:
Why no early return for mismatched classes here?
A `GoalThreeBlocks` and a `GoalTwoBlocks` don't have the same heuristic or goal condition so they aren't equals.